src/openms/extern/tool_description_lib/tdl-config.cmake

```diff
@@ -9,6 +9,19 @@ endif ()
 
 option (INSTALL_TDL "Enable installation of TDL. (Projects embedding TDL may want to turn this OFF.)" ON)
 
+if (ADDRESS_SANITIZER)
+    if (MSVC AND MSVC_VERSION LESS 1920)
+        message(FATAL_ERROR "AddressSanitizer is not supported for MSVC versions < 2019.")
+    else()
+        # For Windows, every static lib that uses STL needs to be compiled with ASAN ON
+        # otherwise the linker will complain about mismatch in annotated vs unannotated STL
+        # types. https://learn.microsoft.com/en-us/cpp/sanitizers/error-container-overflow?view=msvc-170
+        if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+            add_compile_options("/fsanitize=address")
+        endif()
+    endif()
+endif()
```

Comment:
same problem.
Can we add this code after including tdl? i.e. make it part of OpenMS?

Reply:
Yes can do too

Reply:
Yeah so this is a tricky one. Not sure how I get access to the options in fetch content there. Basically what I am doing here is to set the options for all targets generated in that subfolder.

Reply:
If I do it one directory up, it will affect more of our dependencies.
Maybe something like
Add options
Include
Remove options
works from outside.